crates/katana/storage/provider/src/providers/db/mod.rs

```diff
@@ -581,7 +606,9 @@ impl BlockWriter for DbProvider {
             db_tx.put::<tables::Headers>(block_number, block_header)?;
             db_tx.put::<tables::BlockBodyIndices>(block_number, block_body_indices)?;
 
-            for (i, (transaction, receipt)) in transactions.into_iter().zip(receipts).enumerate() {
+            for (i, ((transaction, receipt), execution)) in
+                transactions.into_iter().zip(receipts).zip(executions).enumerate()
```

Comment:
Is the double `zip` ok, or there's a more idiomatic way of doing so?

Reply:
i think its better to assign the adapters into a variable for better readability

```rust
let txs = transactions
    .into_iter()
    .zip(receipts.into_iter())
    .zip(executions.into_iter())
    .map(|((transaction, receipt), execution)| (transaction, receipt, execution))
    .enumerate();

for (i, (transaction, receipt, execution)) in txs { .. } 
```

Reply:
Fixed in 15e63d7ff99fbcc96a67a9e42ca9bb074af62f9a.